### PR TITLE
[SPARK-9398][SQL] Datetime cleanup

### PR DESCRIPTION
**Pull request by yjshen**
JIRA: https://issues.apache.org/jira/browse/SPARK-9398

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala [hunk 1]
```diff
@@ -601,22 +601,51 @@ object DateTimeUtils {
...
    */
   def getLastDayOfMonth(date: Int): Int = {
-    val dayOfMonth = getDayOfMonth(date)
-    val month = getMonth(date)
-    if (month == 2 && isLeapYear(getYear(date))) {
-      date + daysInNormalYear(month - 1) + 1 - dayOfMonth
-    } else {
-      date + daysInNormalYear(month - 1) - dayOfMonth
-    }
+    date + daysToMonthEnd(date)
```
Comment (yjshen) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala]: the previous implementation calls `getDayOfMonth`, `getMonth`, `getYear` consequently, which all internally call `getYearAndDayInYear` at their first line.

**Comment by yjshen** [comment]
Will give another pass once #7718 is merged.

**Comment by rxin** [comment]
cc @davies

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala (hunk 1 above)
Comment (davies) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala]: Could we just put those in `daysToMonthEnd` to here? `daysToMonthEnd` is even harder to understand than `getLastDayOfMonth`

Comment (yjshen) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala]: Sure, I would inline `daysToMonthEnd`.

**Comment by davies** [comment]
LGTM, merging this into master, thanks!

**Pull request closed without merging**